packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -100,6 +100,12 @@ export class CdkToolkit {
       }
     }
 
+    const outputsPath = options.outputsPath;
+    if (outputsPath) {
+      fs.ensureFileSync(outputsPath);
+      fs.writeJSONSync(outputsPath, {}, {encoding: 'utf8'});
```

Comment:
What's up here? why are we writing an empty object first?

Reply:
good question. The reason I ended up doing this was when multiple deploys are made but the same file that was previously provided is used. 

I clear the file before writing to it to avoid confusion between which deployments produced the entries for a provided stack are tied to. It makes sense to simply overwrite if we don't mind the file having outputs from deploys at different times. 

I don't know that we *have* to clear the file we're about to use, but wanted some feedback here. what do you think?

Reply:
cleaned this up by writing in the finally block